cellprofiler/modules/thresholding.py

```diff
@@ -0,0 +1,231 @@
+# coding=utf-8
+
+"""
+
+Thresholding
+
+"""
+
+import cellprofiler.image
+import cellprofiler.module
+import cellprofiler.setting
+import numpy
+import skimage.exposure
+import skimage.filters
+import skimage.morphology
+
+
+class Thresholding(cellprofiler.module.Module):
+    category = "Volumetric"
+    module_name = "Thresholding"
+    variable_revision_number = 1
+
+    def create_settings(self):
+        self.x_name = cellprofiler.setting.ImageNameSubscriber(
+            u"Input"
+        )
+
+        self.y_name = cellprofiler.setting.ImageNameProvider(
+            u"Output",
+            u"OutputImage"
+        )
+
+        self.method = cellprofiler.setting.Choice(
+            u"Method",
+            [
+                u"Adaptive",
+                u"Iterative selection thresholding",
+                u"Manual",
+                u"Minimum cross entropy thresholding",
+                u"Otsu’s method",
+                u"Yen’s method"
+            ]
+        )
+
+        self.adaptive_block_size = cellprofiler.setting.Integer(
+            u"Block size",
+            value=3
+        )
+
+        self.adaptive_method = cellprofiler.setting.Choice(
+            u"Adaptive method",
+            [
+                u"Gaussian",
+                u"Generic",
+                u"Mean",
+                u"Median"
+            ],
+            u"Gaussian"
+        )
+
+        self.adaptive_offset = cellprofiler.setting.Float(
+            u"Offset",
+            value=0.0
+        )
+
+        self.adaptive_mode = cellprofiler.setting.Choice(
```

Comment:
It looks like it's used: https://github.com/CellProfiler/CellProfiler/blob/aae5d0873b7b7c0b607ae9643da63cbcc2fd94f8/cellprofiler/modules/thresholding.py#L174-L180